tests/test_layout_ctors.cpp

```diff
@@ -203,7 +203,8 @@ TEST(TestLayoutLeftListInitialization, test_layout_left_extent_initialization) {
   ASSERT_TRUE(m.is_exhaustive());
 }
 
-#if defined(_MDSPAN_USE_CLASS_TEMPLATE_ARGUMENT_DEDUCTION)
+// FIXME: CUDA NVCC including 12.0 does not like CTAD on nested classes
```

Comment:
It's not really an mdspan FIXME; it's an NVCC FIXME ; - )  Would you consider rewording?

```suggestion
// CUDA NVCC, including 12.0, implements CTAD but
// is not always able to perform CTAD on nested classes.
```

Another option would be to add a macro to config.hpp specifically for this case.  Then, you could comment the macro there once, so you wouldn't have to repeat the comment for each disabled test.

Reply:
The FIXME is just a reminder that we need to revisit this with later versions of CUDA.